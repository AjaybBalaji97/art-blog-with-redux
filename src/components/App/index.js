import React, { Component } from 'react';
import BlogHeader from '../BlogHeader';
import BlogFooter from '../BlogFooter';
import AddPostReaction from '../../containers/AddPostReaction';
import CreatePost from '../CreatePost';
import './App.css';
import getPosts from  '../../helper/getPosts';
let count=0;
class App extends Component {
state={
  axiosPosts:[],
  loading:true,
}

async componentDidMount (){
    //console.log(this.props);
    return await getPosts().then((res)=>{
      this.setState({
        axiosPosts:res,
        
        loading:false
      });
      if(count===0){
      this.state.axiosPosts.map((post)=>{this.props.initializePosts(post)});
      count=1;
      }
      // this.props.initializePosts({title:'Learn about actions',liked:true,claps:3});
    })
      
   
  }
  componentWillUnmount(){
    console.log('unmount');
  }
  render() {
    //console.log(this.state.axiosPosts);
    //console.log(this.props);
    if(this.state.loading){
      return <div>LOADING...</div>
    }
    else{
    return (
     
      <div className="App">
        <BlogHeader/>
        <AddPostReaction posts={this.props.blogs}/>
        <CreatePost history={this.props.history}/>
        <BlogFooter/>
      </div>
    );
    }
  }
}

export default App;
