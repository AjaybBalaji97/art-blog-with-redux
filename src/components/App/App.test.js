import React from 'react';
import renderer from 'react-test-renderer';
import App from '.';
import {shallow,mount} from 'enzyme';
import axios from 'axios';

describe('App',()=>{
    const products=[{
        "date": "2nd Januray, 2018",
        "readingTime": "2 mins",
        "title": "The future of abstract art and the culture ...",
        "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
        "claps": 10,
        "liked": false,
        "image": "abstract.png"
    },{
        "date": "2nd Januray, 2018",
        "readingTime": "2 mins",
        "title": "The future of abstract art and the culture ...",
        "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
        "claps": 10,
        "liked": false,
        "image": "abstract.png"
    }]
    it('should render All Posts ',()=>{
        
        const tree=renderer.create(<App  />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
describe('App',()=>{
    it('componentDidMount should be called',()=>{
        const componentDidMountSpy = jest.spyOn(App.prototype, 'componentDidMount');
        const wrapper=shallow(<App/>);
        expect(App.prototype.componentDidMount).toHaveBeenCalled();
    })
})

describe('"Axios" axios get function', () => {
//     let getMock;
//     const mockPosts = {data:[{
//         userId: 1,
//         id: 1,
//         title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
//         body: 'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto',
//       }]};
//   beforeAll(() => {
//     getMock = jest.spyOn(axios, 'get');
//     getMock.mockImplementation(() => mockPosts);
//   });
//   afterAll(() => {
//     getMock.mockRestore();
//   });
    it('should call componentDidMount', async() => {
        //jest.mock('axios');
        const wrapper= shallow(<App/>);
        const resp = {data: [{cities: ['NY']}]};
        expect(wrapper.instance().state.axiosPosts).toEqual([]);
        //qaxios.get.mockImplementation(() => Promise.resolve(resp));
         await wrapper.instance().componentDidMount()
        expect(wrapper.instance().state.axiosPosts).not.toEqual([]);

    });
  });