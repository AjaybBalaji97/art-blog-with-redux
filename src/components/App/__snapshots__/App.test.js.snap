// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render All Posts  1`] = `
<div>
  LOADING...
</div>
`;
